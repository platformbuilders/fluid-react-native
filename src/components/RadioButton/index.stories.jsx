import { storiesOf } from '@storybook/react-native';
import radioButtonMarkdown from './radioButton.md';

storiesOf('Radio Button', module)
  .addParameters({
    readme: {
      content: radioButtonMarkdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => {});
