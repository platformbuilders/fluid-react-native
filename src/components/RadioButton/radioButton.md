# Radio Button ğŸ“»

## Example

![Text Input](https://user-images.githubusercontent.com/4137064/97174520-f8084680-1770-11eb-9170-f635e3d67bc0.png)

## Source Code

```js
import React from 'react';
import { RadioButton as DefaultRadioButton } from '@platformbuilders/react-native-ui';

const RadioButton: React.FC = () => {
  return (
    <RadioButton
      id="id_radio_button"
      accessibility="Select Information"
      size={18}
      internalSize={13}
      checked
      onPress={(): void => {}}
    />
  );
};

export default RadioButton;
```

## Props

| Name                    | Required | Type     | Default Value | Description                   |
| ----------------------- | -------- | -------- | ------------- | ----------------------------- |
| id                      | true     | string   | ' '           |                               |
| accessibility           | true     | string   | ' '           |                               |
| radioButtonColor        | false    | string   | '#212121'     | border color                  |
| checkedRadioButtonColor | false    | string   | '#212121'     | internal color                |
| checked                 | false    | boolean  | false         | checked / uncheked            |
| size                    | false    | number   | 18            | size of radio button          |
| internalSize            | false    | number   | 13            | internal size of radio button |
| onPress                 | false    | function | (): any => {} | handle press radio button     |
