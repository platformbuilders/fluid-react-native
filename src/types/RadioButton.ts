import { StyleProp, TextStyle } from 'react-native';

export type RadioButtonType = {
  id?: string;
  accessibility: string;
  accessibilityLabel?: string;
  testID?: string;
  radioButtonColor?: string;
  checkedRadioButtonColor?: string;
  size?: number;
  internalSize?: number;
  checked?: boolean;
  disabled?: boolean;
  onPress?(): void;
  label?: string;
  labelStyle?: StyleProp<TextStyle>;
};
