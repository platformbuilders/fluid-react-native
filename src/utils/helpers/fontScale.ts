import { RFValue } from 'react-native-responsive-fontsize';

export const scaledFontSize = (
  size: number | string,
  baselineFactor?: 280,
): number => {
  const value = typeof size === 'string' ? parseInt(size, 10) : size;
  return RFValue(value, baselineFactor);
};
