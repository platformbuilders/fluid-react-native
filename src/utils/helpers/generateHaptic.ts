import Haptic, { HapticFeedbackType } from 'react-native-haptic';
import { isIOS } from './isIOS';

export const generateHaptic = (haptic: HapticFeedbackType = 'impact'): void => {
  if (isIOS()) {
    Haptic.generate(haptic);
  }
};
