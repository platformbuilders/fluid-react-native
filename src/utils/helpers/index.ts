export {
  currencyParser,
  parseToThousands,
  removeWhiteSpaces,
  toOnlyNumbers,
  getTheme,
  ifStyle,
  switchStyle,
} from '@platformbuilders/helpers';
export * from './generateHaptic';
export * from './getByTypographyVariant';
export * from './fontScale';
export * from './shadow';
