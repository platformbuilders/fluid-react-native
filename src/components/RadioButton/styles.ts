import styled from 'styled-components/native';
import { moderateScale } from 'react-native-size-matters';
import { getTheme, ifStyle } from '../../utils/helpers';
import Touchable from '../Touchable';

const textColor = getTheme('text');
const hasColor = ifStyle('radioButtonColor');
const hasCheckedColor = ifStyle('checkedRadioButtonColor');

type RadioProps = {
  radioButtonColor?: string;
  size: number;
};

export const Radio = styled(Touchable)<RadioProps>`
  width: ${({ size }: RadioProps) => moderateScale(size)}px;
  height: ${({ size }: RadioProps) => moderateScale(size)}px;
  border-radius: ${({ size }: RadioProps) => moderateScale(size / 2)}px;
  border: ${moderateScale(1)}px
    ${({ radioButtonColor }: RadioProps): any =>
      hasColor(radioButtonColor, textColor)};
  align-items: center;
  justify-content: center;
`;

type CheckRadioProps = {
  checkedRadioButtonColor?: string;
  internalSize: number;
};

export const CheckedRadio = styled.View<CheckRadioProps>`
  width: ${({ internalSize }: CheckRadioProps) =>
    moderateScale(internalSize)}px;
  height: ${({ internalSize }: CheckRadioProps) =>
    moderateScale(internalSize)}px;
  background-color: ${({ checkedRadioButtonColor }: CheckRadioProps): any =>
    hasCheckedColor(checkedRadioButtonColor, textColor)};
  border-radius: ${({ internalSize }: CheckRadioProps) =>
    moderateScale(internalSize / 2)}px;
`;
