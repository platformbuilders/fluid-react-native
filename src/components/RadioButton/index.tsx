import React from 'react';
import If from '../If';
import { Radio, CheckedRadio } from './styles';

type Props = {
  id?: string;
  accessibility: string;
  radioButtonColor?: string;
  checkedRadioButtonColor?: string;
  size?: number;
  internalSize?: number;
  checked?: boolean;
  onPress?(): void;
};

const RadioButton: React.FC<Props> = ({
  id,
  accessibility,
  radioButtonColor,
  checkedRadioButtonColor,
  size = 18,
  internalSize = 13,
  checked = false,
  onPress = () => {},
}) => (
  <Radio
    id={id || accessibility}
    accessibility={accessibility}
    onPress={onPress}
    radioButtonColor={radioButtonColor}
    size={size}
  >
    <If condition={checked}>
      <CheckedRadio
        checkedRadioButtonColor={checkedRadioButtonColor}
        internalSize={internalSize}
      />
    </If>
  </Radio>
);

export default RadioButton;
