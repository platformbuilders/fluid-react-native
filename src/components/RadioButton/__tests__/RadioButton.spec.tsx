import 'jest';
import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';
import { ThemeProvider } from 'styled-components/native';
import RadioButton from '..';
import { theme } from '../../../test/helpers';

describe('<RadioButton />', () => {
  it('should render radio button component', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <RadioButton
          id="testing_radio_btn"
          accessibility="Select information"
        />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should trigger onPress function', () => {
    const onPressEvent = jest.fn();

    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <RadioButton
          id="testing_radio_btn"
          accessibility="checkbox"
          onPress={onPressEvent}
        />
      </ThemeProvider>,
    );

    const component = getByTestId('testing_radio_btn');

    fireEvent.press(component);

    expect(onPressEvent).toHaveBeenCalled();
  });
});
