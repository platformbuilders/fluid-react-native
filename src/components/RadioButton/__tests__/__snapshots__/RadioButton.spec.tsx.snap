// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> should render radio button component 1`] = `
<View
  accessibilityLabel="Select information"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#212121",
      "borderRadius": 14.142857142857142,
      "borderStyle": "solid",
      "borderWidth": 1.5714285714285714,
      "height": 28.285714285714285,
      "justifyContent": "center",
      "opacity": 1,
      "width": 28.285714285714285,
    }
  }
  testID="testing_radio_btn"
/>
`;
